x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/topmetrics/TopMetricsAggregatorMetricsTests.java

```diff
@@ -203,7 +207,8 @@ private ValuesSourceConfig toConfig(SortedNumericDocValues values) throws IOExce
         when(source.isFloatingPoint()).thenReturn(false);
         when(source.longValues(null)).thenReturn(values);
         if (randomBoolean()) {
-            return toConfig(source, CoreValuesSourceType.NUMERIC, randomWholeNumberDocValuesFormat(), true);
+            // TODO: should we have a case for CVST.LONG here?
+            return toConfig(source, CoreValuesSourceType.DOUBLE, randomWholeNumberDocValuesFormat(), true);
```

Comment:
It's probably worth it now that they are separate types. But it's not a blocker for this at all.